---
title: 'Metagenomics analysis of phages from global urban sewage'
author: "Josephine Strange"
date: "June 2020"
output: 
  html_document:
    theme: sandstone
    code_folding: hide
---

Install previous version (3.2.1)of ggplot2
```{r}
PackageUrl <- "http://cran.r-project.org/src/contrib/Archive/ggplot2/ggplot2_3.2.1.tar.gz"
install.packages(PackageUrl, repos = NULL, type ="source")
```

Load libraries
```{r init, message=FALSE}

# install previous package of ggplot2
# PackageUrl <- "http://cran.r-project.org/src/contrib/Archive/ggplot2/ggplot2_3.2.1.tar.gz"
# install.packages(PackageUrl, repos=NULL, type="source")
# packageVersion("ggplot2")
library(dendsort)
library(tidyverse)
library(RColorBrewer)
library(dplyr)
library(grid)
library(pheatmap)
library(ape)
library(vegan)
library(compositions)
library(ggfortify)
library(viridis)
library(knitr)
library(reshape2)
library(mganalysis) # Patrickmunk/mganalysis from bitbucket
# devtools::install_git("https://bitbucket.org/nicholasehamilton/ggtern")
# library(ggtern)

theme_set(theme_bw())
```

Density - coverage NCBI phages
```{r, eval=TRUE}
# Import all files of a directory
data <- read.delim("/media/phine/Backup/thesis/data/final/pilot/datasets/sequencing_stats/ncbi_coverage_matrix.phages.tsv",header = 1,row = 1)
data.melt = melt(data)
# mean(data.melt$value)
# min(data.melt$value)
# max(data.melt$value)
# sd(data.melt$value)

# Find number of phages above 80% 
data.subset <- Filter(function(x) any(x > 80), data)


# pdf("plots/final/2016_coverage_ncbi.phages.pdf")
ggplot(data.melt, aes(x=value)) +
  geom_density(fill="red") + 
  ggtitle("Coverage distribution of NCBI phages\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5)) +
  xlab("Template identity (%)") + 
  ylab("Density") 
  
# dev.off()

```

Density - coverage IMG/VR genomes
```{r, eval=TRUE}
data <- read.delim("datasets/sequencing_stats/imgvr_coverage_matrix.genomes.tsv",header=1,row=1)
data.melt <- melt(data)

data.subset <- Filter(function(x) any(x >= 80), data)

# mean(data.melt$value)
# min(data.melt$value)
# max(data.melt$value)
# sd(data.melt$value)
# mean = 2.5
# min = 0
# max = 97.62 
# sd = 7,78

# pdf("plots/final/2016_coverage_imgvr.genomes.pdf")
ggplot(data.melt, aes(x=value)) +
  geom_density(fill = "red") +
  ggtitle("Coverage distribution of IMG/VR genomes\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5)) +
  xlab("Template identity (%)") +
  ylab("Density")
# dev.off()
```

Density - coverage huge phages
```{r, eval=TRUE}
data <- read.delim("datasets/sequencing_stats/custom_coverage_matrix.tsv")
data.melt <- melt(data)

data.subset <- Filter(function(x) any(x >= 80), data)

# mean(data.melt$value)
# min(data.melt$value)
# max(data.melt$value)
# sd(data.melt$value)

# pdf("plots/final/2016_coverage_custom.pdf")
ggplot(data.melt, aes(x=value)) +
  geom_density(fill="red") + 
  ggtitle("Coverage distribution of huge phages\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5)) +
  xlab("Template identity (%)") +
  ylab("Density")
# dev.off()
```

Density - depth NCBI phages (from res)
```{r, eval=TRUE}
data <- read.delim("datasets/sequencing_stats/ncbi_depth_matrix.phages.tsv")
data.melt <- melt(data)

data.subset <- Filter(function(x) any(x >= 30), data)

# # mean(data.melt$value)
# min(data.melt$value)
# max(data.melt$value)
# sd(data.melt$value)

# data.subset <- data[,!colMeans(data) < 25]

# pdf("plots/final/2016_depth_ncbi.phages.pdf")
ggplot(data.melt, aes(x=value)) +
  geom_density(fill="red") + 
  ggtitle("Depth distribution of NCBI phages\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5)) +
  xlab("Depth") +
  ylab("Density")
# dev.off()
```

Density - depth IMG/VR genomes (from res)
```{r, eval=TRUE}
data <- read.delim("datasets/sequencing_stats/imgvr_depth_matrix.genomes.tsv",header=1,row=1)
data.melt <- melt(data)

data.subset <- Filter(function(x) any(x >= 30), data)

# mean(data.melt$value)
# min(data.melt$value)
# max(data.melt$value)
# sd(data.melt$value)

# pdf("plots/final/2016_depth_imgvr.genomes.pdf")
ggplot(data.melt, aes(x=value)) +
  geom_density(fill="red") + 
  ggtitle("Depth distribution of IMG/VR genomes\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5)) +
  xlab("Depth") +
  ylab("Density")
# dev.off()

```

Density - depth huge phages (from res)
```{r, eval=TRUE}
data <- read.delim("datasets/sequencing_stats/custom_depth_matrix.tsv",header=1,row=1)
data.melt <- melt(data)

data.subset <- Filter(function(x) any(x >= 30), data)

# mean(data.melt$value)
# min(data.melt$value)
# max(data.melt$value)
# sd(data.melt$value)

# pdf("plots/final/2016_depth_custom.pdf")
ggplot(data.melt, aes(x=value)) +
  geom_density(fill="red") + 
  ggtitle("Depth distribution of huge phages\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5)) +
  xlab("Depth") +
  ylab("Density")
# dev.off()

```

Density - max depth NCBI (from mapstat)
```{r}
data <- read.delim("/media/phine/Backup/thesis/data/depth/ncbi_depth_matrix.mapstat.txt",header=1,row=1)

# What phages have a max depth above 30
data.subset <-Filter(function(x) any(x > 30), data)

# 39 phages 
write.table(data.subset, "/media/phine/Backup/thesis/data/depth/ncbi_depth_30.txt",sep="\t", row.names = T, col.names = NA)

# Convert to long format
data.melt <- melt(data)

# Plot the data
ggplot(data.melt, aes(x=value)) +
  geom_density(fill="red") + 
  ggtitle("Max depth distribution of NCBI phages\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5)) +
  xlab("Max depth") +
  ylab("Density")


```

Density - max depth IMG/VR (from mapstat)
```{r}
data <- read.delim("/media/phine/Backup/thesis/data/depth/imgvr_depth_matrix.genomes.mapstat.txt",header=1,row=1)

# What phages have a max depth above 30
data.subset <-Filter(function(x) any(x > 30), data)
max(data.subset)

# 343 phages 
write.table(data.subset, "/media/phine/Backup/thesis/data/depth/imgvr_depth_30.txt",sep="\t", row.names = T, col.names = NA)

# Convert to long format
data.melt <- melt(data)

# plot
ggplot(data.melt, aes(x=value)) +
  geom_density(fill="red") + 
  ggtitle("Max depth distribution of NCBI phages\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5)) +
  xlab("Max depth") +
  ylab("Density")

```

Density - max depth huge phages (from mapstat)
```{r}
data <- read.delim("/media/phine/Backup/thesis/data/depth/custom_depth_matrix.mapstat.txt",header=1,row=1)

# What phages have a max depth above 30
data.subset <-Filter(function(x) any(x > 30), data)
max(data.subset)

# 22 phages 
write.table(data.subset, "/media/phine/Backup/thesis/data/depth/custom_depth_30.txt",sep="\t", row.names = T, col.names = NA)

# Convert to long format
data.melt <- melt(data)

# plot
ggplot(data.melt, aes(x=value)) +
  geom_density(fill="red") + 
  ggtitle("Max depth distribution of NCBI phages\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5)) +
  xlab("Max depth") +
  ylab("Density")
```

Mapped summary fragment count - all
```{r, eval=TRUE}
# Read data to table
data = read.table("datasets/summary/summary_fragmentCount.tsv", sep='\t', dec='.', header=TRUE, check.names=FALSE)

levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Defining key names
names(data) <- c("sample", "region", "NCBI: phages", "NCBI: other viruses", "IMG/VR: phages", "IMG/VR: other viruses", "KVIT: phages", "KVIT: other viruses", "Huge phages", "Bacteria") 

# Colour pallette
pal <- colorRampPalette(brewer.pal(7,"Paired"))(7) 
colors_ms = c(pal, "grey") # Bacteria in gray

# Reshape data
data_m = melt(data)

# Plot data with bars
# pdf('plots/final/2016_fragmentCount.pdf',onefile=TRUE, width=15, height=10)

ggplot(data=data_m,aes(x=sample,y=value,fill=variable)) + 
  geom_bar(stat = "identity") + 
  scale_fill_manual(values=colors_ms,name="Database") +
  ylab("Fragment count") + 
  xlab("") +
  ggtitle("Fragment count summary - all\n2016") +
  theme(plot.title = element_text(face="bold", hjust=0.5)) +
  theme(axis.text.x = element_text(size=9,angle=270,hjust=0, vjust=1)) +
  facet_grid(.~region,scales="free",space="free") +
  theme(strip.text.x = element_text(size=12),
        strip.background = element_blank())

# dev.off()

# Calculate mean and SD
mean <- mean(data_m$value)
```

Mapped summary fragment abundance - all
```{r, eval=TRUE}

data = read.table("datasets/summary/summary_relabundance.tsv", sep='\t', dec='.', header=TRUE, check.names=FALSE)

levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

names(data) <- c("sample","region","NCBI: phages", "NCBI: other viruses", "IMG/VR: phages", "IMG/VR: other viruses", "KVIT: phages", "KVIT: other viruses", "Huge phages", "Bacteria") ## Dataset renaming

pal <- colorRampPalette(brewer.pal(7,"Paired"))(7) ## Creating pallette for different datasets

pal


colors_ms = c(pal, "grey") # Bacteria is grey
data_m = melt(data) # Reshape the data

myvars <- names(data) %in% c("sample","region","KVIT: phages")

# Find the mean relative abundance 
num.data <- t(data[!myvars])
mean_rows <- apply(num.data,1,mean)
mean <- mean(mean_rows)

sd_rows <- apply(num.data,1,sd)

# plot with bars
pdf('plots/final/2016_relativeReadAbundance.pdf', onefile=TRUE,width=15, height=10)

ggplot(data=data_m, 
       aes(x=sample, 
           y=value, 
           fill=variable)) + 
  geom_bar(stat = "identity") + 
  scale_fill_manual(values=colors_ms,name="Database") + 
  ylab("Mapped fragment fraction (%)") + 
  xlab("") + 
  ggtitle("Relative abundance summary\n2016") + 
  theme(plot.title = element_text(lineheight=.8,face="bold",hjust = 0.5)) + 
  theme(axis.text.x = element_text(size=9,angle=270,hjust=0,vjust=1)) + 
  facet_grid(.~variable) +
  facet_grid(.~region,
             scales="free",
             space="free") + 
  theme(strip.text.x = element_text(size=12), 
        strip.background = element_blank())

dev.off()

```

Mapped summary readCount - all viruses
```{r, eval=TRUE}
# Data load
data = read.table("datasets/summary/summary_fragmentCount_viruses.tsv", sep='\t', dec='.', header=TRUE, check.names=FALSE)

levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Key labels
names(data) <- c("sample","region","NCBI: phages", "NCBI: other viruses", "IMG/VR: phages", "IMG/VR: other viruses", "KVIT: phages", "KVIT: other viruses", "Huge phages")

# Colour pallette
pal <- colorRampPalette(brewer.pal(7,"Paired"))(7) 
data_m = melt(data) # Shape the data

# plot with bars 
pdf('plots/final/2016_fragmentCount.viruses.pdf',onefile=TRUE, width=15, height=10)

ggplot(data=data_m, aes(x=sample, y=value, fill=variable)) + 
  geom_bar(stat = "identity") + 
  scale_fill_manual(values=pal, name="Database") + 
  ylab("Fragment count") + 
  xlab("") +
  ggtitle("Fragment count summary - phages and viruses\n2016") + 
  theme(plot.title = element_text(lineheight=.8, face="bold", hjust = 0.5)) +
  theme(axis.text.x = element_text(size=9, angle=270, hjust=0, vjust=1)) +
  facet_grid(.~region,scales="free",space="free") + 
  theme(strip.text.x = element_text(size=12),
        strip.background = element_blank())
dev.off()
```

Mapped summary read abundance - all viruses
```{r, eval=TRUE}
data = read.table("datasets/summary/summary_relabundance_viruses.tsv", sep='\t', dec='.', header=TRUE, check.names=FALSE)

# Update region label
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

names(data) <- c("sample","region", "NCBI: phages", "NCBI: other viruses", "IMG/VR: phages", "IMG/VR: other viruses", "KVIT: phages", "KVIT: other viruses", "Huge phages") 

myvars <- names(data) %in% c("sample","region","KVIT: phages")

# Find the mean relative abundance 
num.data <- data[!myvars]
mean_rows <- apply(num.data,1,mean)
mean <- mean(mean_rows)
sd <- sd(mean_rows)

# Colouring pallette
pal <- colorRampPalette(brewer.pal(7,"Paired"))(7)

# Reshape data
data_m = melt(data)

# plot with bars
pdf('plots/final/2016_relativeReadAbundance.viruses.pdf', width=15, height=10)

ggplot(data=data_m, aes(x=sample, y=value, fill=variable)) + 
  geom_bar(stat = "identity") + 
  scale_fill_manual(values=pal, name="Database") + 
  ylab("Mapped fragment fraction (%)") + 
  xlab("") + 
  ggtitle("Relative abundance summary - phages and viruses\n2016") + 
  theme(plot.title = element_text(lineheight=.8, face="bold", hjust = 0.5)) + 
  theme(axis.text.x = element_text(size=9, angle=270, hjust=0, vjust=1)) +
  facet_grid(.~region,scales="free",space="free") + 
  theme(strip.text.x = element_text(size=12), 
        strip.background = element_blank())

dev.off()
```

Regional distributions - NCBI
```{r}
data <- read.delim("datasets/virus_id_relabundance.phages.tsv",header=1,row=1)

# Group data by region and find sum for each phage
data.grouped <- aggregate(data[,2:ncol(data)], list(data$region), sum)

# Set region to rowname
rownames(data.grouped) <- data.grouped[,1]

data.grouped <- data.frame(t(data.grouped[,2:ncol(data.grouped)]))

tot_AF <- nrow(data.grouped[data.grouped$Africa > 0,])
tot_AS <- nrow(data.grouped[data.grouped$Asia > 0,])
tot_EU <- nrow(data.grouped[data.grouped$Europe > 0,])
tot_ME <- nrow(data.grouped[data.grouped$Middle.East > 0,])
tot_NAm <- nrow(data.grouped[data.grouped$North.America > 0,])
tot_OC <- nrow(data.grouped[data.grouped$Oceania > 0,])
tot_SA <- nrow(data.grouped[data.grouped$South.America > 0,])

# disregard categorical column
tot <- ncol(data[,2:ncol(data)])

pctAF <- tot_AF / tot * 100
pctAS <- tot_AS / tot * 100
pctEU <- tot_EU / tot * 100
pctME <- tot_ME / tot * 100
pctNAm <- tot_NAm / tot * 100
pctOC <- tot_OC / tot * 100
pctSA <- tot_SA / tot * 100


# write.table(data.grouped, "ncbi_region_grouped.txt",sep="\t",dec=",",row.names=T,col.names=T)

```

Regional distributions - IMG/VR
```{r}
# data <- read.delim("datasets/imgvr_fragments/imgvr_id_relabundance.phages.tsv",header=1,row=1)
data <- read.delim("datasets/imgvr_fragments/imgvr_id_relabundance.phages.tsv", header = 1, row = 1) # Fragments
# data <- read.delim("datasets/imgvr_genomes/imgvr_id_relabundance.phages.tsv", header = 1, row = 1) # Genomes

# Group data by region and find sum for each phage
data.grouped <- aggregate(data[,3:ncol(data)], list(data$region), sum)

# Set region to rowname
rownames(data.grouped) <- data.grouped[,1]

data.grouped <- data.frame(t(data.grouped[,2:ncol(data.grouped)]))

tot_AF <- nrow(data.grouped[data.grouped$Africa > 0,])
tot_AS <- nrow(data.grouped[data.grouped$Asia > 0,])
tot_EU <- nrow(data.grouped[data.grouped$Europe > 0,])
tot_ME <- nrow(data.grouped[data.grouped$Middle.East > 0,])
tot_NAm <- nrow(data.grouped[data.grouped$North.America > 0,])
tot_OC <- nrow(data.grouped[data.grouped$Oceania > 0,])
tot_SA <- nrow(data.grouped[data.grouped$South.America > 0,])

# disregard categorical column
tot <- ncol(data[,2:ncol(data)])

pctAF <- tot_AF / tot * 100
pctAS <- tot_AS / tot * 100
pctEU <- tot_EU / tot * 100
pctME <- tot_ME / tot * 100
pctNAm <- tot_NAm / tot * 100
pctOC <- tot_OC / tot * 100
pctSA <- tot_SA / tot * 100

# write.table(data.grouped, "imgvr_genome_region_grouped.txt",sep="\t",dec=",",row.names=T,col.names=T)
```

Regional distributions - huge phages
```{r}
data <- read.delim("datasets/custom_id_relabundance.tsv",header = 1, row = 1)

#Create tables wiht regional mean abundance
# Group data by region and find means for each phage
data.grouped <- aggregate(data[,3:ncol(data)], list(data$region), sum)

# Set region to rowname
rownames(data.grouped) <- data.grouped[,1]


data.grouped <- data.frame(t(data.grouped[,2:ncol(data.grouped)]))


write.table(data.grouped, "custom_region_grouped.txt",sep="\t",dec=",",row.names=T,col.names=T)
```

PCoA - NCBI phages
```{r, eval=TRUE}
# Read input 
data <- read.delim("datasets/virus_id_relabundance.phages.tsv",header = 1, row = 1)

levels(data.bc$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data[,2:ncol(data)]

#Calculate dissimilarity matrix
data.dist <- vegdist(decostand(data.num ,method = "hellinger"), dist="bray")

# Get region label from dataframe
metadata <- data.frame(sample=rownames(data.num), data.num, group=c(rep("Africa",13), rep("Asia",9), rep("Europe",31), rep("Middle East",3), rep("North America",14), rep("Oceania",3), rep("South America",5))) 

# PCoA plot with Ellipse
betadisp.all <- betadisper(data.dist, data$region, add=F)

eigenperc <- round(as.vector(eigenvals(betadisp.all) / sum(eigenvals(betadisp.all)[eigenvals(betadisp.all)>0]))*100, digits=2)

num.var = length(unique(data$region))

# Setup output and plot to file
# pdf('plots/final/2016_PCoA_NCBI.phages.pdf',onefile=TRUE)
plot(betadisp.all,
     ellipse = T,
     hull = F,
     col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     pch=rep(16,num.var),
     lty=1,
     lwd=2,
     segments=T,
     seg.col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     label=T,
     cex=1.5, 
     main="Distribution of other NCBI viruses\n2016",
     seg.lty=2, seg.lwd=1, xlab=paste("PCoA1 (", eigenperc[1] , "% variation)"), ylab=paste("PCoA2 (", eigenperc[2] , "% variation)")) 
# dev.off()
```

PCoA - NCBI viruses
```{r, eval=TRUE}
# Read input 
data = read.delim("datasets/virus_id_relabundance.virus.tsv", header=1, row = 1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data[,2:ncol(data)]

#Calculate dissimilarity matrix
data.dist <- vegdist(decostand(data.num ,method = "hellinger"), dist="bray")

# Get region label from dataframe
metadata <- data.frame(sample=rownames(data.num), data.num, group=c(rep("Africa",13), rep("Asia",9), rep("Europe",31), rep("Middle East",3), rep("North America",14), rep("Oceania",3), rep("South America",5))) 

# PCoA plot with Ellipse
betadisp.all <- betadisper(data.dist, data$region, add=F)

eigenperc <- round(as.vector(eigenvals(betadisp.all) / sum(eigenvals(betadisp.all)[eigenvals(betadisp.all)>0]))*100, digits=2)

num.var = length(unique(data$region))

# Setup output and plot to file
# pdf('plots/final/2016_PCoA_NCBI.virus.pdf',onefile=TRUE)
plot(betadisp.all,
     ellipse = T,
     hull = F,
     col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     pch=rep(16,num.var),
     lty=1,
     lwd=2,
     segments=T,
     seg.col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     label=T,
     cex=1.5, 
     main="Distribution of other NCBI viruses\n2016",
     seg.lty=2, seg.lwd=1, xlab=paste("PCoA1 (", eigenperc[1] , "% variation)"), ylab=paste("PCoA2 (", eigenperc[2] , "% variation)")) 
# dev.off()
```

PCoA - NCBI bacteria
```{r, eval=TRUE}
# Read input 
data = read.delim("datasets/bacteria_species_name_relabundance.tsv", header = 1, row = 1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data[,2:ncol(data)]

#Calculate dissimilarity matrix
data.dist <- vegdist(decostand(data.num ,method = "hellinger"), dist="bray")

# Get region label from dataframe
metadata <- data.frame(sample=rownames(data.num), data.num, group=c(rep("Africa",13), rep("Asia",9), rep("Europe",31), rep("Middle East",3), rep("North America",14), rep("Oceania",3), rep("South America",5))) 

# PCoA plot with Ellipse
betadisp.all <- betadisper(data.dist, data$region, add=F)

eigenperc <- round(as.vector(eigenvals(betadisp.all) / sum(eigenvals(betadisp.all)[eigenvals(betadisp.all)>0]))*100, digits=2)

num.var = length(unique(data$region))

# Setup output and plot to file
# pdf('plots/final/2016_PCoA_NCBI.bacteria.pdf',onefile=TRUE,width=11)
plot(betadisp.all, 
     ellipse = T, 
     hull = F,
     xlim=c(-0.4,0.2),
     col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     pch=rep(16,num.var),
     lty=1, 
     lwd=3,
     segments=T, 
     seg.col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"),
     label=T, 
     cex=1.5, 
     main="Distribution of NCBI bacteria\n2016",
     seg.lty=2, 
     seg.lwd=1,
     xlab=paste("PCoA1 (", eigenperc[1] , "% variation)"),
     ylab=paste("PCoA2 (", eigenperc[2] , "% variation)"))
# dev.off()
```

PCoA - IMG/VR phages
```{r , eval=TRUE}
# Read input 
data = read.delim("datasets/imgvr_fragments/imgvr_id_relabundance.phages.tsv", header = 1, row = 1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data[,2:ncol(data)]

#Calculate dissimilarity matrix
data.dist <- vegdist(decostand(data.num ,method = "hellinger"), dist="bray")

# Get region label from dataframe
metadata <- data.frame(sample=rownames(data.num), data.num, group=c(rep("Africa",13), rep("Asia",9), rep("Europe",31), rep("Middle East",3), rep("North America",14), rep("Oceania",3), rep("South America",5))) 

# PCoA plot with Ellipse
betadisp.all <- betadisper(data.dist, data$region, add=F)

eigenperc <- round(as.vector(eigenvals(betadisp.all) / sum(eigenvals(betadisp.all)[eigenvals(betadisp.all)>0]))*100, digits=2)

num.var = length(unique(data$region))

# Setup output and plot to file
# pdf('plots/final/2016_PCoA_IMGVR.phage.pdf',onefile=TRUE)

plot(betadisp.all, 
     ellipse = T, 
     hull = F, 
     col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     pch=rep(16,num.var), 
     lty=1, 
     lwd=2, 
     segments=T, 
     seg.col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     label=T, cex=1.5, 
     main="Distribution of IMG/VR phages (incl. fragments)\n2016", 
     seg.lty=2, 
     seg.lwd=1, 
     xlab=paste("PCoA1 (", eigenperc[1] , "% variation)"), 
     ylab=paste("PCoA2 (", eigenperc[2] , "% variation)")) 
# dev.off()
```

PCoA - IMG/VR phage genomes
```{r, eval=TRUE}
# Read input 
data = read.delim("datasets/imgvr_genomes/imgvr_id_relabundance.phages.tsv", header = 1, row = 1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data[,2:ncol(data)]

#Calculate dissimilarity matrix
data.dist <- vegdist(decostand(data.num ,method = "hellinger"), dist="bray")

# Get region label from dataframe
metadata <- data.frame(sample=rownames(data.num), data.num, group=c(rep("Africa",13), rep("Asia",9), rep("Europe",31), rep("Middle East",3), rep("North America",14), rep("Oceania",3), rep("South America",5))) 

# PCoA plot with Ellipse
betadisp.all <- betadisper(data.dist, data$region, add=F)

eigenperc <- round(as.vector(eigenvals(betadisp.all) / sum(eigenvals(betadisp.all)[eigenvals(betadisp.all)>0]))*100, digits=2)

num.var = length(unique(data$region))

# Setup output and plot to file
# pdf('plots/final/2016_PCoA_IMGVR.phage_genomes.pdf',onefile=TRUE)
plot(betadisp.all, 
     ellipse = T, 
     hull = F, 
     col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     pch=rep(16,num.var), 
     lty=1, 
     lwd=2, 
     segments=T, 
     seg.col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     label=T, cex=1.5, 
     main="Distribution of IMG/VR phage genomes\n2016", 
     seg.lty=2, 
     seg.lwd=1, 
     xlab=paste("PCoA1 (", eigenperc[1] , "% variation)"), 
     ylab=paste("PCoA2 (", eigenperc[2] , "% variation)")) 
# dev.off()
```

PCoA - IMG/VR other viruses
```{r, eval=TRUE}
# Read input 
data = read.delim("datasets/imgvr_fragments/imgvr_id_relabundance.virus.tsv", header = 1, row = 1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data[,2:ncol(data)]

#Calculate dissimilarity matrix
data.dist <- vegdist(decostand(data.num ,method = "hellinger"), dist="bray")

# Get region label from dataframe
metadata <- data.frame(sample=rownames(data.num), data.num, group=c(rep("Africa",13), rep("Asia",9), rep("Europe",31), rep("Middle East",3), rep("North America",14), rep("Oceania",3), rep("South America",5))) 

# PCoA plot with Ellipse
betadisp.all <- betadisper(data.dist, data$region, add=F)

eigenperc <- round(as.vector(eigenvals(betadisp.all) / sum(eigenvals(betadisp.all)[eigenvals(betadisp.all)>0]))*100, digits=2)

num.var = length(unique(data$region))

# Setup output and plot to file
# pdf('plots/final/2016_PCoA_IMGVR.virus.pdf',onefile=TRUE)
plot(betadisp.all, ellipse = T, 
     hull = F, 
     col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     pch=rep(16,num.var), 
     lty=1, 
     lwd=2, 
     segments=T, 
     seg.col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     label=T, 
     cex=1.5, 
     main="Distribution of other IMG/VR viruses\n2016", 
     seg.lty=2, 
     seg.lwd=1, 
     xlab=paste("PCoA1 (", eigenperc[1] , "% variation)"), 
     ylab=paste("PCoA2 (", eigenperc[2] , "% variation)"))
# dev.off()

```

PCoA - Huge phages
```{r, eval=TRUE}
# Read input 
data = read.delim("datasets/custom_id_relabundance.tsv", header = 1, row = 1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data[,2:ncol(data)]

#Calculate dissimilarity matrix
data.dist <- vegdist(decostand(data.num ,method = "hellinger"), dist="bray")

# Get region label from dataframe
metadata <- data.frame(sample=rownames(data.num), data.num, group=c(rep("Africa",13), rep("Asia",9), rep("Europe",31), rep("Middle East",3), rep("North America",14), rep("Oceania",3), rep("South America",5))) 

# PCoA plot with Ellipse
betadisp.all <- betadisper(data.dist, data$region, add=F)

eigenperc <- round(as.vector(eigenvals(betadisp.all) / sum(eigenvals(betadisp.all)[eigenvals(betadisp.all)>0]))*100, digits=2)

num.var = length(unique(data$region))

# Setup output and plot to file
# pdf('plots/final/2016_PCoA_huge.phages.pdf',onefile=TRUE)
plot(betadisp.all, 
     ellipse = T,
     hull = F, 
     col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     pch=rep(16,num.var),
     lty=1, 
     lwd=2,
     segments=T, 
     seg.col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"),
     label=T, 
     cex=1.5,
     main="Distribution of huge phages\n2016",
     seg.lty=2, 
     seg.lwd=1, 
     xlab=paste("PCoA1 (", eigenperc[1] , "% variation)"), 
     ylab=paste("PCoA2 (", eigenperc[2] , "% variation)")) 
# dev.off()

```

PCoA - concatenated phages
```{r,eval=TRUE}
# Read input 
data = read.delim("datasets/pcoa/phages.concatenated.tsv", header = 1, row = 1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data[,2:ncol(data)]

#Calculate dissimilarity matrix
data.dist <- vegdist(decostand(data.num ,method = "hellinger"), dist="bray")

# Get region label from dataframe
metadata <- data.frame(sample=rownames(data.num), data.num, group=c(rep("Africa",13), rep("Asia",9), rep("Europe",31), rep("Middle East",3), rep("North America",14), rep("Oceania",3), rep("South America",5))) 

# PCoA plot with Ellipse
betadisp.all <- betadisper(data.dist, data$region, add=F)

eigenperc <- round(as.vector(eigenvals(betadisp.all) / sum(eigenvals(betadisp.all)[eigenvals(betadisp.all)>0]))*100, digits=2)

num.var = length(unique(data$region))

# Setup output and plot to file
# pdf('plots/final/2016_PCoA_concatenated.phages.pdf',onefile=TRUE)
plot(betadisp.all, 
     ellipse = T, 
     hull = F, 
     col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     pch=rep(16,num.var), 
     lty=1, 
     lwd=2, 
     segments=T, 
     seg.col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"),
     label=T,
     cex=1.5, 
     main="Distribution of all phages\n2016",
     seg.lty=2,
     seg.lwd=1, 
     xlab=paste("PCoA1 (", eigenperc[1] , "% variation)"),
     ylab=paste("PCoA2 (", eigenperc[2] , "% variation)")) 
# dev.off()
```

PCoA - concatenated phages - IMG/VR genomes - Not used
```{r,eval=TRUE}
# Read input 
data = read.delim("datasets/pcoa/phages.concatenated.ncbi_custom_imgvrGenomes.tsv",header = 1, row = 1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data[,2:ncol(data)]

#Calculate dissimilarity matrix
data.dist <- vegdist(decostand(data.num ,method = "hellinger"), dist="bray")

# Get region label from dataframe
metadata <- data.frame(sample=rownames(data.num), data.num, group=c(rep("Africa",13), rep("Asia",9), rep("Europe",31), rep("Middle East",3), rep("North America",14), rep("Oceania",3), rep("South America",5))) 

# PCoA plot with Ellipse
betadisp.all <- betadisper(data.dist, data$region, add=F)

eigenperc <- round(as.vector(eigenvals(betadisp.all) / sum(eigenvals(betadisp.all)[eigenvals(betadisp.all)>0]))*100, digits=2)

num.var = length(unique(data$region))

# Setup output and plot to file
# pdf('plots/final/2016_PCoA_concatenated.phages_ncbi_custom_imgvrGenomes.pdf',onefile=TRUE)
plot(betadisp.all, 
     ellipse = T, 
     hull = F, 
     col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     pch=rep(16,num.var), 
     lty=1, 
     lwd=2, 
     segments=T, 
     seg.col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"),
     label=T,
     cex=1.5, 
     main="Distribution of all phages\n2016",
     seg.lty=2,
     seg.lwd=1, 
     xlab=paste("PCoA1 (", eigenperc[1] , "% variation)"),
     ylab=paste("PCoA2 (", eigenperc[2] , "% variation)")) 
# dev.off()
```

PCoA - concatenated other viruses
```{r, eval=TRUE}
# Read input 
data = read.delim("datasets/pcoa/virus.concatenated.tsv", header = 1, row = 1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data[,2:ncol(data)]

#Calculate dissimilarity matrix
data.dist <- vegdist(decostand(data.num ,method = "hellinger"), dist="bray")

# Get region label from dataframe
metadata <- data.frame(sample=rownames(data.num), data.num, group=c(rep("Africa",13), rep("Asia",9), rep("Europe",31), rep("Middle East",3), rep("North America",14), rep("Oceania",3), rep("South America",5))) 

# PCoA plot with Ellipse
betadisp.all <- betadisper(data.dist, data$region, add=F)

eigenperc <- round(as.vector(eigenvals(betadisp.all) / sum(eigenvals(betadisp.all)[eigenvals(betadisp.all)>0]))*100, digits=2)

num.var = length(unique(data$region))

# Setup output and plot to file
# pdf('plots/final/2016_PCoA_concatenated.viruses.pdf',onefile=TRUE)
plot(betadisp.all,
     ellipse = T,
     hull = F, 
     col=c("#E41A1C","#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"), 
     pch=rep(16,num.var), 
     lty=1, 
     lwd=2, 
     segments=T, 
     seg.col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"),
     label=T,
     cex=1.5,
     main="Distribution of all other viruses\n2016", 
     seg.lty=2, seg.lwd=1, 
     xlab=paste("PCoA1 (", eigenperc[1] , "% variation)"),
     ylab=paste("PCoA2 (", eigenperc[2] , "% variation)")) 
# dev.off()

```

PCoA - ResFinder - genes
```{r, eval=TRUE}
# Read data
data = read.delim("datasets/resistance/resfinder_genes_FPKM.tsv",header = 1, row = 1)  
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data[,2:ncol(data)]

#Calculate dissimilarity matrix
data.dist <- vegdist(decostand(data.num ,method = "hellinger"), dist="bray")

# Get region label from dataframe
metadata <- data.frame(sample=rownames(data.num), data.num, group=c(rep("Africa",13), rep("Asia",9), rep("Europe",31), rep("Middle East",3), rep("North America",14), rep("Oceania",3), rep("South America",5))) 

# PCoA plot with Ellipse
betadisp.all <- betadisper(data.dist, data$region, add=F)

eigenperc <- round(as.vector(eigenvals(betadisp.all) / sum(eigenvals(betadisp.all)[eigenvals(betadisp.all)>0]))*100, digits=2)

num.var = length(unique(data$region))

# Write to PDF
# pdf('plots/final/2016_PCoA_resfinder.genes.FPKM.pdf',onefile=TRUE)
plot(betadisp.all,
     ellipse = T,
     hull = F,
     col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"),
     pch=rep(16,num.var),
     lty=1,
     lwd=2,
     segments=T,
     seg.col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"),
     label=T,
     cex=1,
     main="Distribution of ResFinder genes - FPKM\n2016",
     seg.lty=2,
     seg.lwd=1,
     xlab=paste("PCoA1 (", eigenperc[1] , "% variation)"),
     ylab=paste("PCoA2 (", eigenperc[2] , "% variation)"))
# dev.off()

```

PCoA Functional Resistance - genes
```{r, eval=TRUE}
# Read data
data = read.delim("datasets/resistance/functionalResistance_genes_FPKM.tsv", header = 1, row = 1)  
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data[,2:ncol(data)]

#Calculate dissimilarity matrix
data.dist <- vegdist(decostand(data.num ,method = "hellinger"), dist="bray")

# Get region label from dataframe
metadata <- data.frame(sample=rownames(data.num), data.num, group=c(rep("Africa",13), rep("Asia",9), rep("Europe",31), rep("Middle East",3), rep("North America",14), rep("Oceania",3), rep("South America",5))) 

# PCoA plot with Ellipse
betadisp.all <- betadisper(data.dist, data$region, add=F)

eigenperc <- round(as.vector(eigenvals(betadisp.all) / sum(eigenvals(betadisp.all)[eigenvals(betadisp.all)>0]))*100, digits=2)

num.var = length(unique(data$region))

# Write to PDF
# pdf('plots/final/2016_PCoA_functionalResistance.genes.FPKM.pdf',onefile=TRUE)
plot(betadisp.all,
     ellipse = T,
     hull = F,
     col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"),
     pch=rep(16,num.var),
     lty=1,
     lwd=2,
     segments=T,
     seg.col=c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"),
     label=T,
     cex=1.5,
     main="Distribution of Functional Resistance genes - FPKM\n2016",
     seg.lty=2,
     seg.lwd=1,
     xlab=paste("PCoA1 (", eigenperc[1] , "% variation)"),
     ylab=paste("PCoA2 (", eigenperc[2] , "% variation)"))
# dev.off()

```

Ternary plot - NCBI phages - relabundances
```{r, eval=TRUE}
data <- read.delim("datasets/income/virus_id_relabundance.phages.tsv")

# Group data by income and find means for each phage
data.grouped <- aggregate(data[,3:ncol(data)], list(data$income_class), mean)

# Set rownames to income-class
rownames(data.grouped) <- data.grouped[,1]

# Exclude non-numerical column and transpose data frame
data.grouped <- data.frame(t(data.grouped[,2:ncol(data.grouped)]))

# Annotate the phages with a number
data.grouped$phage_num <- c(1:nrow(data.grouped))

# Plot the data
# pdf("plots/final/2016_ternary_ncbi.phages.mid_zoom.pdf")
ggtern(data.grouped, aes(x = high, y = middle, z = low)) +
  # stat_confidence_tern(geom="polygon", aes(fill=..level..), color="white") +
  geom_point_swap(fill = "gray50", stat = "identity", alpha=0.4,shape=16, size = 5) +
  geom_density_tern(geom="polygon", bdl=0.001) + 
  # breaks_tern()
  
   # Labels
  labs(title="Phage distribution stratified by income class\nNCBI virus 2016") +
  xlab("High-income") +
  ylab("Middle-income") +
  zlab("Low-income") +
  geom_text(aes(label=phage_num),size=2) +

  
  # Theme tweaks
  theme_rgbw() +
  theme_showarrows() +
  theme(plot.title = element_text(face="bold",hjust = .5)) +
  theme(tern.axis.title.L = element_text(hjust = 0,vjust=2),
            tern.axis.title.R = element_text(hjust = 1,vjust=2)) 

  # Zoom functions
  # tern_limits(X=.25,L=1,R=.025,)
   # theme_zoom_L(0.5)  # zoom on high-income
   # theme_zoom_R(0.5)   # zoom on low-income
  # theme_zoom_T(0.5)  # zoom on middle-income
# dev.off()

# write.table(data.grouped, "ncbi_tern.xlsx",sep="\t",row.names=T,col.names=T)
```

Ternary plot - IMG/VR genomes - relabundances
```{r, eval=TRUE}
data <- read.delim("datasets/income/imgvr_id_relabundance.phages.tsv",header=1,row=1)

# Group data by income and find means for each phage
data.grouped <- aggregate(data[,3:ncol(data)], list(data$income_class), mean)

# Set rownames to income-class
rownames(data.grouped) <- data.grouped[,1]

# Exclude non-numerical column and transpose data frame
data.grouped <- data.frame(t(data.grouped[,2:ncol(data.grouped)]))

# Plot the data 
# pdf("plots/final/2016_ternary_imgvr.genomes.pdf")
# Annotate the phages with a number
data.grouped$phage_num <- c(1:nrow(data.grouped))

# Plot the data
pdf("plots/final/2016_ternary_imgvr.genomes.mid_zoom.pdf")
ggtern(data.grouped, aes(x = high, y = middle, z = low)) +
  # stat_confidence_tern(geom="polygon", aes(fill=..level..), color="white") +
  geom_point_swap(fill = "gray50", stat = "identity", alpha=0.4,shape=16, size = 5) +
  
   # Labels
  labs(title="Phage distribution stratified by income class\nIMG/VR genomes 2016") +
  xlab("High-income") +
  ylab("Middle-income") +
  zlab("Low-income") +
  geom_text(aes(label=phage_num),size=2) +
  
  # Theme tweaks
  theme_rgbw() +
  theme_showarrows() +
  theme(plot.title = element_text(face="bold",hjust = .5)) +
  theme(tern.axis.title.L = element_text(hjust = 0,vjust=2),
            tern.axis.title.R = element_text(hjust = 1,vjust=2)) +

  # Zoom functions
  # theme_zoom_L(0.5) + # zoom on high-income
  # theme_zoom_R(0.5) +  # zoom on low-income
  # theme_zoom_T(0.5) + # zoom on middle-income
    geom_mask()
dev.off()

# WRite table to save annotations
# write.table(data.grouped, "imgvr_tern.xlsx",sep="\t",row.names=T,col.names=T)
```

Ternary plot - huge phages - relabundances
```{r, eval=TRUE}
data <- read.delim("datasets/income/custom_id_relabundance.tsv",header=1,row=1)

# Group data by income and find means for each phage
data.grouped <- aggregate(data[,3:ncol(data)], list(data$income_class), mean)

# Set rownames to income-class
rownames(data.grouped) <- data.grouped[,1]

# Exclude non-numerical column and transpose data frame
data.grouped <- data.frame(t(data.grouped[,2:ncol(data.grouped)]))

# Annotate the phages with a number
data.grouped$phage_num <- c(1:nrow(data.grouped))

# Plot the data
pdf("plots/final/2016_ternary_huge_phages.low_zoom.pdf")
ggtern(data.grouped, aes(x = high, y = middle, z = low)) +
  # stat_confidence_tern(geom="polygon", aes(fill=..level..), color="white") +
  geom_point_swap(fill = "gray50", stat = "identity", alpha=0.4,shape=16, size = 5) +
  
   # Labels
  labs(title="Phage distribution stratified by income class\nHuge phages 2016") +
  xlab("High-income") +
  ylab("Middle-income") +
  zlab("Low-income") +
  geom_text(aes(label=phage_num),size=2) +
  
  # Theme tweaks
  theme_rgbw() +
  theme_showarrows() +
  theme(plot.title = element_text(face="bold",hjust = .5)) +
  theme(tern.axis.title.L = element_text(hjust = 0,vjust=2),
            tern.axis.title.R = element_text(hjust = 1,vjust=2)) +

  # Zoom functions
  # theme_zoom_L(0.5) + # zoom on high-income
  theme_zoom_R(0.5) +  # zoom on low-income
  # theme_zoom_T(0.5) + # zoom on middle-income
    geom_mask()
dev.off()

# write.table(data.grouped, "custom_tern.xlsx",sep="\t",row.names=T,col.names=T)

```

Ternary plot - NCBI - distributions 
```{r}
data <- read.delim("datasets/income/distributions.txt",header=1,row=1)


ggtern(data, aes(x=distribution_low, y=distribution_middle, z=distribution_high)) +
  geom_point_swap(fill = "gray50", stat = "identity", alpha=0.4,shape=16, size = 5) +
  
   # Labels
  labs(title="Phage distribution stratified by income class\nHuge phages 2016") +
  xlab("High-income") +
  ylab("Middle-income") +
  zlab("Low-income") +
  geom_text(aes(label=phage_number),size=2) +
  
  # Theme tweaks
  theme_rgbw() +
  theme_showarrows() +
  theme(plot.title = element_text(face="bold",hjust = .5)) +
  theme(tern.axis.title.L = element_text(hjust = 0,vjust=2),
            tern.axis.title.R = element_text(hjust = 1,vjust=2)) #+

  # Zoom functions
  # theme_zoom_L(0.5) + # zoom on high-income
  # theme_zoom_R(0.5) +  # zoom on low-income
  # theme_zoom_T(0.5) + # zoom on middle-income
```

Heatmap top 15 per sample - NCBI top phages
```{r}
data <- read.delim("datasets/top/virus_id_relabundance.phages.top15_sample.tsv", header = 1, row = 1)

levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

data.num <- data[,2:ncol(data)]

data.bc <- vegan::vegdist(decostand(data.num, "hellinger"), method = "bray")
data.pear <- dist(cor(data.num, method = "pearson"))

# Make annotation data frame linking sample to region
region.df = data.frame("Region" = data[,"region"])
rownames(region.df) <- rownames(data.num)

# Color intensities using quantile breaks
quantile_breaks <- function(mat, n = 11){
  breaks <- quantile(mat, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}

breaklist <- quantile_breaks(t(data.num), n = 200)

# Dendrogram sorting
sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
cluster.col <- sort_hclust(hclust(data.bc))
cluster.row <- sort_hclust(hclust(data.pear))

region.cols = list(
  Region = c("AF" = "#E41A1C",
             "AS" = "#377EB8",
             "EU" = "#4DAF4A",
             "ME" = "#984EA3",
             "NA" = "#FF7F00",
             "OC" = "#FFFF33",
             "SA" = "#A65628"))

# Remove dots from names
names(data.num) <- gsub("\\.", " ", names(data.num))

# Heatmap plotting
pdf("2016_pheatmap_NCBI_phages.top15_sample.raw.pdf", height = 32, width = 16)
pheatmap(mat = t(data.num),
         
         # Coloration and intensities
         color = inferno(length(breaklist) - 1),
         breaks = breaklist,
         legend = T,         
         
         # Clustering
         cluster_rows = cluster.row,
         cluster_cols = cluster.col,
         
         # Annotations
         annotation_col = region.df,
         annotation_colors = region.cols,
         show_rownames = T,
         
         # Cell dimensions
         cellwidth = 10,
         cellheight = 10,
         
         # Plot name
         main = "Relative abundances of  NCBI phages\nTop 15 per sample\n2016")
dev.off()

```

Heatmap top 15 per sample - IMG/VR genomes
```{r}
data <- read.delim("datasets/top/imgvr_id_relabundance.phages.top15_sample.tsv", header = 1, row = 1)
# Update region levels
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")
# Subset to numerical data
data.num <- data[,2:ncol(data)]
# Calculate distances
data.bc <- vegan::vegdist(decostand(data.num, "hellinger"), method = "bray")
data.pear <- dist(cor(data.num, method = "pearson"))
# Make annotation data frame linking sample to region
region.df = data.frame("Region" = data[,"region"])
rownames(region.df) <- rownames(data.num)
# Color intensities using quantile breaks
quantile_breaks <- function(mat, n = 11){
  breaks <- quantile(mat, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}

breaklist <- quantile_breaks(t(data.num), n = 15)

# Column dendrogram sorting
sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
cluster.col <- sort_hclust(hclust(data.bc))
cluster.row <- sort_hclust(hclust(data.pear))

# Region coloration
region.cols = list(
  Region = c("AF" = "#E41A1C",
             "AS" = "#377EB8",
             "EU" = "#4DAF4A",
             "ME" = "#984EA3",
             "NA" = "#FF7F00",
             "OC" = "#FFFF33",
             "SA" = "#A65628"))

# Heatmap plotting
pdf("2016_pheatmap_imgvr_genomes.top15_sample.raw.pdf", height = 31, width = 20)
pheatmap(mat = t(data.num),
         
         # Coloration and intensities
         color = inferno(10),
         breaks = breaklist,
         legend = T,         
         
         # Clustering
         cluster_rows = cluster.row,
         cluster_cols = cluster.col,
         
         # Annotations
         annotation_col = region.df,
         annotation_colors = region.cols,
         show_rownames = T,

         cellwidth = 10,
         cellheight = 10,
         main = "Relative abundances of IMG/VR genomes\nTop 15 per sample\n2016")
dev.off()

```

Heatmap top 15 per sample - huge phages
```{r}
data <- read.delim("datasets/top/custom_id_relabundance.top15_sample.tsv", header = 1, row = 1)
# Update region levels
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")
# Subset to numerical data
data.num <- data[,2:ncol(data)]
# Distance matrix for sample clustering 
data.bc <- vegan::vegdist(decostand(data.num, "hellinger"), method = "bray")
data.pear <- dist(cor(data.num, method = "pearson"))
# Log transform data
data.log <- log(data.num + 1)
# Remove dots in names
names(data.log) <- gsub("\\.", " ", names(data.log))
# Make annotation data frame linking sample to region
region.df = data.frame("Region" = data[,"region"])
rownames(region.df) <- rownames(data.num)

# Quantile coloration and intensities
quantile_breaks <- function(mat, n = 11){
  breaks <- quantile(mat, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}

breaklist <- quantile_breaks(t(data.num), n = 30)

# Region annotation
region.cols = list(
  Region = c("AF" = "#E41A1C",
             "AS" = "#377EB8",
             "EU" = "#4DAF4A",
             "ME" = "#984EA3",
             "NA" = "#FF7F00",
             "OC" = "#FFFF33",
             "SA" = "#A65628"))

# Dendrogram sorting on BC matrix
sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
cluster.col <- sort_hclust(hclust(data.bc))
cluster.row <- sort_hclust(hclust(data.pear))

# Heatmap plotting
pdf("2016_custom.top15_sample.raw.pdf", height = 15, width = 17)
pdf("test.pdf", height = 31, width = 16)
pheatmap(mat = t(data.num),
         
         # Coloration and intensities
         color = inferno(length(breaklist) - 1),
         breaks = breaklist,
         legend = T,         
         
         # Clustering and ordering rows/columns
         cluster_cols = cluster.col,
         cluster_rows = cluster.row,
         
         # Annotations
         annotation_col = region.df,
         annotation_colors = region.cols,
         show_rownames = T,

         cellwidth = 10,
         cellheight = 10,
         main = "Relative abundances of huge phages\nTop 15 per sample\n2016")
dev.off()

```

Bacterial top 15 per sample - NCBI bacteria genus
```{r}
data <- read.delim("datasets/top/bacteria_genus_name_relabundance.top15_sample.tsv", row = 1, header = 1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data[,2:ncol(data)]

# BC distance for clustering
data.bc <- vegdist(decostand(data.num, "hellinger"), dist = "bray")
data.pear <- dist(cor(data.num, method = "pearson"))

# Log transform data
data.log <- log10(data.num + 1)

# Remove dots from variables
names(data.log) <- gsub("\\.", " ", names(data.log))

# Annotation and coloration scheme
region.df = data.frame("Region" = data[,"region"])
rownames(region.df) <- rownames(data)
region.cols = list(
  Region = c("AF" = "#E41A1C",
             "AS" = "#377EB8",
             "EU" = "#4DAF4A",
             "ME" = "#984EA3",
             "NA" = "#FF7F00",
             "OC" = "#FFFF33",
             "SA" = "#A65628"))

# Color intensities using quantile breaks
quantile_breaks <- function(mat, n = 11){
  breaks <- quantile(mat, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}

breaklist <- quantile_breaks(t(data.log), n = 20)

# Dendrogram sorting on BC matrix and pearson dist objects
sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
cluster.col <- sort_hclust(hclust(data.bc))
cluster.row <- sort_hclust(hclust(data.pear))

# Plotting
pdf("plots/2016_pheatmap_bacteria_genus.log.pdf",width = 15, height = 15)
pheatmap(mat = t(data.log),
         color = inferno(length(breaklist) - 1),
         clustering_distance_rows = "correlation", # Pearson correlation
         
         # Dendrogram arrangement; does not change clustering just orders them
         cluster_col = cluster.col,
         cluster_rows = cluster.row,
         
         # Annotations
         annotation_col = region.df,
         annotation_colors = region.cols,
         breaks = breaklist,
         
         cellheight = 10,
         cellwidth = 10,
         main = "Log-transformed relative abundances of NCBI bacterial genera\nTop 15 per sample\n2016"
         )
dev.off()

```

Heatmap - NCBI phage families - relative abundance, cluster BC
```{r, eval=TRUE}
data <- read.delim("datasets/virus_id_relabundance.phages.families.tsv",header=1,row=1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Sort data on region 
data <- data[order(data$region),]

# Exclude non-numerical data 
data.num <- data.frame(data[,2:ncol(data)], row.names=rownames(data))

# Hellinger transform and calculate BC dissimilarity matrix + pearson correlation 
data.bc <- vegdist(decostand(data.num, "hellinger"), dist="bray") 
data.pear <- dist(cor(data.num, method = "pearson"))

# Region annotation
region.df = data.frame("Region" = data[,"region"])
rownames(region.df) <- rownames(data)

# Coloration scheme 
region.cols = list(
  Region = c("AF" = "#E41A1C",
             "AS" = "#377EB8",
             "EU" = "#4DAF4A",
             "ME" = "#984EA3",
             "NA" = "#FF7F00",
             "OC" = "#FFFF33",
             "SA" = "#A65628"))

# Quantile coloration and intensities
quantile_breaks <- function(mat, n = 11){
  breaks <- quantile(mat, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}

breaklist <- quantile_breaks(t(data.num), n = 30)

# Dendrogram sorting on BC matrix
sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
cluster.col <- sort_hclust(hclust(data.bc))
cluster.row <- sort_hclust(hclust(data.pear))

# Plot heatmap
pdf("plots/2016_pheatmap_NCBI.phage_families.pdf",width=20,height=10)
# pdf("plots/final/2016_pheatmap_NCBI.phage_families.pdf",width=20,height=10)
pheatmap(mat = t(data.num),
         # Coloration and intensities
         color = inferno(5),
         breaks = breaklist,
         # Clustering
         cluster_col = cluster.col,
         cluster_rows = cluster.row,
         # Annotation
         annotation_col = region.df,
         annotation_colors = region.cols,
         # Cell dimensions
         cellheight = 10,
         cellwidth = 10,
         main = "Relative abundances of NCBI phage families\n2016"
         )
dev.off()

```

Heatmap - Huge phages - relative abundances - clades, cluster bc
```{r, eval=TRUE}
# Read input 
data <- read.delim("datasets/custom_id_relabundance.clades.tsv",header=1,row=1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")
# Subset to numerical data
data.num <- data.frame(data[,2:ncol(data)], row.names=rownames(data))

# BC dissimilarities on Hellinger standardised relative abundances
data.bc <- vegdist(decostand(data.num, "hellinger"), dist="bray") 
data.pear <- dist(cor(data.num, method = "pearson"))
# Update display name
names(data.num) <- gsub("\\.", " ",names(data.num))

# Region annotation
region.df = data.frame("Region" = data[,"region"])
rownames(region.df) <- rownames(data)

# Coloration scheme 
region.cols = list(
  Region = c("AF" = "#E41A1C",
             "AS" = "#377EB8",
             "EU" = "#4DAF4A",
             "ME" = "#984EA3",
             "NA" = "#FF7F00",
             "OC" = "#FFFF33",
             "SA" = "#A65628"))


# Quantile coloration and intensities
quantile_breaks <- function(mat, n = 11){
  breaks <- quantile(mat, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}

breaklist <- quantile_breaks(t(data.num), n = 30)

# Dendrogram sorting on BC matrix
sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
cluster.col <- sort_hclust(hclust(data.bc))
cluster.row <- sort_hclust(hclust(data.pear))

# Plot heatmap
pdf("plots/final/2016_pheatmap_custom.clades.pdf",width=20,height=10)
pheatmap(mat = t(data.num),
         # Coloration
         color = inferno(length(breaklist) - 1),
         breaks = breaklist,
         # Clustering
         cluster_col= cluster.col,
         cluster_rows = cluster.row,
         # Annotation
         annotation_col = region.df,
         annotation_colors = region.cols,
         # Cell dimensions
         cellheight = 10,
         cellwidth = 10,
         main = "Relative abundances clades of huge phages\n2016"
         )
dev.off()

``` 

Heatmap - Huge phages - relative abundances - host phylum
```{r, eval=TRUE}
# Read input 
data <- read.delim("datasets/custom_id_relabundance.host_phyla.tsv",header=1,row=1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")
# Subset to numerical data
data.num <- data.frame(data[,2:ncol(data)], row.names=rownames(data))

# Hellinger transform and calculate BC dissimilarity matrix and Pearson correlation
data.bc <- vegdist(decostand(data.num, "hellinger"), dist="bray")
data.pear <- dist(cor(data.num, method = "pearson"))

# Quantile coloration and intensities
quantile_breaks <- function(mat, n = 11){
  breaks <- quantile(mat, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}

breaklist <- quantile_breaks(t(data.num), n = 10)

# Dendrogram sorting on BC matrix
sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
cluster.col <- sort_hclust(hclust(data.bc))
cluster.row <- sort_hclust(hclust(data.pear))

# Region annotation
region.df = data.frame("Region" = data[,"region"])
rownames(region.df) <- rownames(data)

# Coloration scheme 
region.cols = list(
  Region = c("AF" = "#E41A1C",
             "AS" = "#377EB8",
             "EU" = "#4DAF4A",
             "ME" = "#984EA3",
             "NA" = "#FF7F00",
             "OC" = "#FFFF33",
             "SA" = "#A65628"))

# Plot heatmap
pdf("plots/final/2016_pheatmap_custom.host_phylum.pdf",width=20,height=10)
pheatmap(mat = t(data.num),
         # Coloration
         color = inferno(length(breaklist) - 1),
         breaks = breaklist,
         # Clustering
         cluster_col = cluster.col,
         cluster_rows = cluster.row,
         # Annotation 
         annotation_col = region.df,
         annotation_colors = region.cols,
         # Cell dimensions
         cellheight = 10,
         cellwidth = 10,
         main = "Relative abundances of host phyla of huge phages\n2016"
         )
dev.off()

``` 

Heatmap - Functional Resistance classes - FPKM
```{r, eval=TRUE}
data <- read.delim("datasets/resistance/functionalresistance_class_FPKM.tsv",header=1,row=1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data
data.num <- data.frame(data[,2:ncol(data)], row.names=rownames(data))

# Add pseudocount to 0 count fields
data.num[data.num == 0] <- 1

data.bc <- vegdist(decostand(data.num, "hellinger"), method="bray")
data.pear <- dist(cor(data.num, method = "pearson"))

# Determine max fragment count for each class
maxcount <- apply(data.num, 2, max)

# Log transform subset data
data.ln <- log(data.num)

# Data frame linking region to sample
region.df = data.frame("Region" = data[,"region"])
rownames(region.df) <- rownames(data.num)

# List of region colors
region.cols = list(
  Region = c("AF" = "#E41A1C",
             "AS" = "#377EB8",
             "EU" = "#4DAF4A",
             "ME" = "#984EA3",
             "NA" = "#FF7F00",
             "OC" = "#FFFF33",
             "SA" = "#A65628"))

# Dendrogram sorting on dists
sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
cluster.col <- sort_hclust(hclust(data.bc))
cluster.row <- sort_hclust(hclust(data.pear))

# Plot data
pdf("plots/final/2016_pheatmap_functionalResistance.class.log.pdf",width=20)
pheatmap(mat = t(data.ln),
         # Color intensity
         color = inferno(10),
         
         # Clustering
         cluster_cols = cluster.col,
         cluster_rows = cluster.row,
         
         # Annotation
         annotation_col = region.df,
         annotation_colors = region.cols,
         
         # Cell dimensions
         cellwidth = 10,
         cellheight = 10,
         )
dev.off()
```

Heatmap - ResFinder classes FPKM ln transformed
```{r, eval=TRUE}
data <- read.delim("datasets/resistance/resfinder_genes_FPKM.class.tsv",header=1,row=1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Subset to numerical data 
data.num <- data.frame(data[,2:ncol(data)], row.names=rownames(data))

# Calculate BC
dist.bc <- vegan::vegdist(decostand(data.num, "hellinger"), method = "bray")

# Log transform 
data.ln <- log(data.num + 1)

# Exclude data below threshold
data.ln <- data.ln[, !colMeans(data.ln) < 0.25]

# Prettify AMR class names
names(data.ln) <- gsub("\\.", "/", names(data.ln))

# Make annotation data frame linking sample to region
region.df = data.frame("Region" = data[,"region"])
rownames(region.df) <- rownames(data.num)

# List of regional colors
region.cols = list(
  Region = c("AF" = "#E41A1C",
             "AS" = "#377EB8",
             "EU" = "#4DAF4A",
             "ME" = "#984EA3",
             "NA" = "#FF7F00",
             "OC" = "#FFFF33",
             "SA" = "#A65628"))

# Plot data
pdf("plots/2016_pheatmap_ResFinder_lnFPKM.class.pdf",width=20)
pheatmap(mat = t(data.ln),
         color = inferno(10),
         cluster_rows = T,
         clustering_distance_cols = dist.bc,
         clustering_distance_rows = "correlation", # PEarson correlation
         annotation_col = region.df,
         annotation_colors = region.cols,
         show_rownames = T,
         cutree_cols = 7,
         cellwidth = 10,
         cellheight = 10
         )
dev.off()
```

Procrustes: NCBI phage to AMR
```{r, eval=TRUE}
library(mganalysis)
# According to documentation, needs transposed matrices;
phage.relab.data <- readAbunData("datasets/Procrustes/virus_id_relabundance.phages.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/resfinder_genes_FPKM.transposed.tsv")

phage.dist <- matrix2BCdist(phage.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
procrustes.dist <-dist2procrustes(phage.dist,res.dist)
# Correlation coefficient: 0.8488
# Significance: 0.001

```

Procrustes: Huge phages to AMR
```{r, eval=TRUE}
library(mganalysis)
# According to documentation, needs transposed matrices;
phage.relab.data <- readAbunData("datasets/Procrustes/custom_id_relabundance.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/resfinder_genes_FPKM.transposed.tsv")

phage.dist <- matrix2BCdist(phage.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
procrustes.dist <-dist2procrustes(phage.dist,res.dist)
# Correlation coefficient: 0.7379
# Significance: 0.001
```

Procrustes: IMGVR fragments to AMR
```{r, eval=TRUE}
library(mganalysis)
# According to documentation, needs transposed matrices;
phage.relab.data <- readAbunData("datasets/Procrustes/imgvr_fragments/imgvr_id_relabundance.phages.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/resfinder_genes_FPKM.transposed.tsv")

phage.dist <- matrix2BCdist(phage.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
procrustes.dist <-dist2procrustes(phage.dist,res.dist)
# Correlation: 0.9079
# Significance: 0.001

```

Procrustes: IMGVR genome to AMR
```{r, eval=TRUE}
# According to documentation, needs transposed matrices;
phage.relab.data <- readAbunData("datasets/Procrustes/imgvr_genome/imgvr_id_relabundance.phages.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/resfinder_genes_FPKM.transposed.tsv")

phage.dist <- matrix2BCdist(phage.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
procrustes.dist <- dist2procrustes(phage.dist,res.dist)
# Correlation: 0.875
# Significance: 0.001
```

Procrustes: NCBI phage to bacteria
```{r, eval=TRUE}
# According to documentation, needs transposed matrices;
phage.relab.data <- readAbunData("datasets/Procrustes/virus_id_relabundance.phages.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/bacteria_species_name_relabundance.transposed.tsv")

phage.dist <- matrix2BCdist(phage.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
procrustes.dist <- dist2procrustes(phage.dist,res.dist)
# Correlation: 0.8148
# Significance: 0.001
```

Procrustes: IMGVR fragments to bacteria
```{r, eval=TRUE}
# According to documentation, needs transposed matrices;
phage.relab.data <- readAbunData("datasets/Procrustes/imgvr_fragments/imgvr_id_relabundance.phages.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/bacteria_species_name_relabundance.transposed.tsv")

phage.dist <- matrix2BCdist(phage.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
procrustes.dist <- dist2procrustes(phage.dist,res.dist)
# Correlation: 0.9097
# Significance: 0.001
```

Procrustes: IMGVR genome to bacteria
```{r ,eval=TRUE}
# According to documentation, needs transposed matrices;
phage.relab.data <- readAbunData("datasets/Procrustes/imgvr_genome/imgvr_id_relabundance.phages.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/bacteria_species_name_relabundance.transposed.tsv")

phage.dist <- matrix2BCdist(phage.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
procrustes.dist <- dist2procrustes(phage.dist,res.dist)
# Correlation: 0.8956
# Significance: 0.001
```

Procrustes: Huge phages to bacteria
```{r, eval=TRUE}
# According to documentation, needs transposed matrices;
phage.relab.data <- readAbunData("datasets/Procrustes/custom_id_relabundance.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/bacteria_species_name_relabundance.transposed.tsv")

phage.dist <- matrix2BCdist(phage.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
procrustes.dist <- dist2procrustes(phage.dist,res.dist)
# Correlation: 0.7054
# Significance: 0.001
```

Procrustes: Bacteria species to AMR
```{r, eval=TRUE}
# According to documentation, needs transposed matrices;
bac.relab.data <- readAbunData("datasets/Procrustes/bacteria_species_name_relabundance.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/resfinder_genes_FPKM.transposed.tsv")

# Create distance matrices
bac.dist <- matrix2BCdist(bac.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
dist2procrustes(bac.dist,res.dist)
# Correlation: 0.8813
# Significance: 0.001

``` 

Procrustes: Bacteria genus to AMR
```{r, eval=TRUE}
# According to documentation, needs transposed matrices;
bac.relab.data <- readAbunData("datasets/Procrustes/bacteria_genus_name_relabundance.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/resfinder_genes_FPKM.transposed.tsv")

# Create distance matrices
bac.dist <- matrix2BCdist(bac.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
dist2procrustes(bac.dist,res.dist)
# Correlation: 0.8753
# Significance: 0.001
``` 

Procrustes: NCBI virus to AMR
```{r, eval=TRUE}
# According to documentation, needs transposed matrices;
bac.relab.data <- readAbunData("datasets/Procrustes/virus_id_relabundance.virus.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/resfinder_genes_FPKM.transposed.tsv")

# Create distance matrices
bac.dist <- matrix2BCdist(bac.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
dist2procrustes(bac.dist,res.dist)
# Correlation: 0.7619
# Significance: <0.001
``` 

Procrustes: NCBI virus to bacteria
```{r, eval=TRUE}
# According to documentation, needs transposed matrices;
bac.relab.data <- readAbunData("datasets/Procrustes/virus_id_relabundance.virus.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/bacteria_species_name_relabundance.transposed.tsv")

# Create distance matrices
bac.dist <- matrix2BCdist(bac.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
dist2procrustes(bac.dist,res.dist)
# Correlation: 0.7167
# Significance: <0.001
``` 

Procrustes: IMG/VR other viruses to bacteria
```{r, eval=TRUE}
# According to documentation, needs transposed matrices;
imgvr.relab.data <- readAbunData("datasets/Procrustes/imgvr_id_relabundance.other_virus.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/bacteria_species_name_relabundance.transposed.tsv")

# Create distance matrices
imgvr.dist <- matrix2BCdist(imgvr.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
dist2procrustes(imgvr.dist,res.dist)
# Correlation: 0.8896
# Significance: 0.001
```

Procrustes: IMG/VR other viruses to ARGs
```{r, eval=TRUE}
# According to documentation, needs transposed matrices;
bac.relab.data <- readAbunData("datasets/Procrustes/imgvr_id_relabundance.other_virus.transposed.tsv")
resFinder.fpkm.data <- readAbunData("datasets/Procrustes/resfinder_genes_FPKM.transposed.tsv")

# Create distance matrices
bac.dist <- matrix2BCdist(bac.relab.data)
res.dist <- matrix2BCdist(resFinder.fpkm.data)

# Feed distance matrices to dist2procrustes
dist2procrustes(bac.dist,res.dist)
# Correlation: 0.9033
# Significance: 0.001
```

Distribution of relative abundances dist
```{r, eval=TRUE}
# Read data
imgvr.all.dist <- read.table("datasets/distribution/imgvr_id_relabundance.phages_frags.distribution.tsv",sep="\t",dec=".",header = T)

imgvr.gen.dist <- read.table("datasets/distribution/imgvr_id_relabundance.phages_genomes.distribution.tsv",sep="\t",dec=".",header = T)

virus.dist <- read.table("datasets/distribution/virus_id_relabundance.phages.distribution.tsv",sep="\t",dec=".",header=T)

huge.dist <- read.delim("datasets/distribution/custom_id_relabundance.distribution.tsv",header=1)

bac.dist <- read.table("datasets/distribution/bacteria_species_name_relabundance.distribution.tsv",sep="\t",dec=".",header=T)

# Plot 'em
# IMG/VR all: #B2DF8A
level.order <- c("(0.0, 1e-07]","(1e-07, 5e-07]","(5e-07, 1e-06]","(1e-06, 5e-06]","(5e-06, 1e-05]", "(1e-05, 5e-05]","(5e-05, 0.0001]","(0.0001, 0.0005]","(0.0005, 0.001]","(0.001, 0.005]","(0.005, 0.01]","(0.01, 0.05]","(0.05, 0.1]","(0.1, 0.5]","(0.5, 1.0]")

p1 <- ggplot(imgvr.all.dist, aes(x=factor(X,level=level.order),y=counts)) +
  geom_bar(stat="identity",fill="#B2DF8A") +
  theme_bw() +
  theme(axis.text.x = element_text(angle=45,hjust=1)) +
  xlab("") +
  ylab("Counts") +
  ggtitle("Relative abundance distribution - IMG/VR all\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5))
  
p2 <- ggplot(imgvr.gen.dist, aes(x=factor(X,level=level.order),y=counts)) +
  geom_bar(stat="identity",fill="#B2DF8A") +
  theme_bw() +
  theme(axis.text.x = element_text(angle=45,hjust=1)) +
  xlab("") +
  ylab("") +
  ggtitle("Relative abundance distribution - IMG/VR genomes\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5))

# Virus: #A6CEE3
p3 <- ggplot(virus.dist, aes(x=factor(X,level=level.order),y=counts)) +
  geom_bar(stat="identity",fill="#A6CEE3") +
  theme_bw() +
  theme(axis.text.x = element_text(angle=45,hjust=1)) +
  xlab("") +
  ylab("Counts") +
  ggtitle("Relative abundance distribution - NCBI phages\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5))

p4 <- ggplot(huge.dist, aes(x=factor(X,level=level.order),y=counts)) +
  geom_bar(stat="identity",fill="#FDBF6F") +
  theme_bw() +
  theme(axis.text.x = element_text(angle=45,hjust=1)) +
  xlab("") +
  ylab("") +
  ggtitle("Relative abundance distribution - huge phages\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5))

# Bacteria
p5 <- ggplot(bac.dist, aes(x=factor(X,level=level.order),y=counts)) +
  geom_bar(stat="identity",fill="grey") +
  theme_bw() +
  theme(axis.text.x = element_text(angle=45,hjust=1)) +
  xlab("") +
  ylab("Counts") +
  ggtitle("Relative abundance distribution - NCBI bacteria\n2016") +
  theme(plot.title = element_text(face="bold",hjust = .5))

# # Plot all in one
pdf('plots/final/supplementary/2016_abundance_distributions.pdf', onefile=TRUE, width = 15, height = 25)
vplayout <- function(x, y) viewport(layout.pos.row = x, layout.pos.col = y)
grid.newpage()
pushViewport(viewport(layout = grid.layout(3, 2))) # define the number of rows and columns, where 1 number is row and second is column
print(p1,vp = vplayout(1, 1))
print(p2,vp = vplayout(1, 2))
print(p3,vp = vplayout(2, 1))
print(p4,vp = vplayout(2, 2))
print(p5,vp = vplayout(3, 1))
dev.off()

```

Rarefaction curve NCBI
``` {r, eval = TRUE}
data <- read.delim("datasets/virus_id_fragmentCount.phages.tsv",header=1,row=1)
levels(data$region) = c("AF","AS","EU","ME","NA","OC","SA")

data.num <- data.frame(data[,2:ncol(data)],row.names = rownames(data))

# Observed number of species
S <- specnumber(data.num)

# Minimum sample count to which we are rarefying
raremax <- min(rowSums(data.num))

# Color matrix
col = c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628")
pars <- expand.grid(col = col, stringsAsFactors=F)

# Retrieve regional groups and set coloration from col list
grp <- factor(data$region)
cols <- col[grp]

# Expected species richness in random subsamples of size raremax
Srare <- rarefy(data.num, raremax)
min(Srare) # 8.334244

pdf("plots/final/2016_rarecurve_NCBI.phages.5k.pdf",onefile =T)
rarecurve(data.num, step=20, col=cols ,cex=0.6, label = F, xlim=c(0,5000))
dev.off()

```

Rarefaction curve IMG/VR genomes
``` {r, eval = TRUE}
data <- read.delim("datasets/imgvr_genomes/imgvr_id_fragmentCount.phages.tsv",header=1,row=1)
levels(data$region) = c("AF","AS","EU","ME","NA","OC","SA")

data.num <- data.frame(data[,2:ncol(data)],row.names = rownames(data))

# Observed number of species
S <- specnumber(data.num)

# Minimum sample count to which we are rarefying
raremax <- min(rowSums(data.num))

# Color matrix
col = c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628")
pars <- expand.grid(col = col, stringsAsFactors=F)

# Retrieve regional groups and set coloration from col list
grp <- factor(data$region)
cols <- col[grp]

# Expected species richness in random subsamples of size raremax
Srare <- rarefy(data.num, raremax)
min(Srare) # 236.5812

pdf("plots/final/2016_rarecurve_IMGVR.phage_genomes.zoom.pdf",onefile =T)
rarecurve(data.num, step=20, col=cols ,cex=0.6, label = F,xlim=c(0,20000))
dev.off()
```

Rarefaction curve huge phages
```{r, eval = TRUE}
data <- read.delim("datasets/custom_id_fragmentCount.tsv",header=1,row=1)
levels(data$region) = c("AF","AS","EU","ME","NA","OC","SA")

data.num <- data.frame(data[,2:ncol(data)],row.names = rownames(data))

# Observed number of species
S <- specnumber(data.num)

# Minimum sample count to which we are rarefying
raremax <- min(rowSums(data.num))

# Color matrix
col = c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628")
pars <- expand.grid(col = col, stringsAsFactors=F)

# Retrieve regional groups and set coloration from col list
grp <- factor(data$region)
cols <- col[grp]

# Expected species richness in random subsamples of size raremax
Srare <- rarefy(data.num, raremax)
min(Srare) # 8.334244

pdf("plots/final/2016_rarecurve_custom.phages.full.pdf",onefile =T)
rarecurve(data.num, step=20, col=cols ,cex=0.6, label = F)#,xlim=c(0,1000))
dev.off()
```

Boxplots - NCBI rarefied diversity, richness and evenness
```{r, eval=TRUE}
data <- read.delim("datasets/virus_id_fragmentCount.phages.tsv",header=1,check.names = F,row.names=1)
levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Exclude non-numerical data and transpose the matrix
myvars <- names(data) %in% c("region")
data.num <- data[!myvars]

# Convert to matrix
data.num <- as.matrix(data.num)

# Rarefy data
(raremax <- min(rowSums(data.num))) # 140
data.rar <- rrarefy(data.num,sample=raremax)

levels(data$region) <- c("AF","AS","EU","ME","NA","OC","SA")

col <- rep(c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FFFF33", "#A65628"),c(12,10,31,3,14,3,5))

# max(J) # 0.3782794S
#### Boxplots #### 
# Calculate Simpson diversity 
simpson <- data.frame(diversity(data.rar,index="simpson"))
simpson$region <- data$region

p1 <- ggplot(simpson, aes(x=region,y=simpson$diversity,fill=region)) +
  geom_boxplot(show.legend=F,outlier.shape = NA) +
  geom_jitter(show.legend = F,shape=21) +
  ggtitle("Simpson diversity\nNCBI - phages\n2016") +
  theme(plot.title = element_text(lineheight=.8, face="bold", hjust = 0.5)) +
  xlab("") + 
  ylab("Index") +
  scale_fill_manual(values=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#FFFF33","#A65628"))

# Richness 
# Calculate Chao1 richness
richness <- data.frame(estimateR(data.rar)[2,])
richness$region <- data$region

p2 <- ggplot(richness, aes(x=region, y=richness$estimate,fill=region)) +
  geom_boxplot(show.legend=F, outlier.shape = NA) +
  geom_jitter(show.legend=F,shape=21) +
  ggtitle("Chao1 richness\nNCBI - phages\n2016") +
  theme(plot.title = element_text(lineheight=.8, face="bold", hjust = 0.5)) +
  xlab("") + 
  ylab("") +
  scale_fill_manual(values=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#FFFF33","#A65628"))

# Pileou evenness
S <- apply(data.rar[,-1]>0,1,sum)
J <- data.frame(diversity(data.rar,index="simpson")/log(S))
# min(J) # 0.1030062
J$region <- data$region

p3 <- ggplot(J, aes(x=region,y=J$diversity,fill=region)) +
  geom_boxplot(show.legend=F, outlier.shape = NA) +
  geom_jitter(show.legend=F,shape=21) +
  ggtitle("Pileou evenness\nNCBI - phages\n2016") +
  theme(plot.title = element_text(lineheight=.8, face="bold", hjust = 0.5)) +
  xlab("") + 
  ylab("") +
  scale_fill_manual(values=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#FFFF33","#A65628"))

# # Plot all in one
# pdf('plots/final/ncbi_boxplot_rarefied.pdf', onefile=TRUE ,width = 20)
vplayout <- function(x, y) viewport(layout.pos.row = x, layout.pos.col = y)
grid.newpage()
pushViewport(viewport(layout = grid.layout(1, 3)))
print(p1,vp = vplayout(1, 1))
print(p2,vp = vplayout(1, 2))
print(p3,vp = vplayout(1, 3))
# dev.off()
```

Boxplots - IMG/VR rarefied diversity, richness and evenness
```{r, eval=TRUE}
imgvr.dat <- read.delim("datasets/imgvr_genomes/imgvr_id_fragmentCount.phages.tsv",header=1,check.names = F,row.names=1)

levels(imgvr.dat$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Exclude non-numerical data
myvars <- names(imgvr.dat) %in% c("region")
imgvr.num <- imgvr.dat[!myvars]

# Convert to matrix 
imgvr.num <- as.matrix(imgvr.num)

# Rarefy data 
(raremax <- min(rowSums(imgvr.num))) #4013
imgvr.rar <- rrarefy(imgvr.num,sample=raremax)

### Boxplots ###
col <- c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#FFFF33","#A65628")

# Calculate Simpson diversity 
simpson <- data.frame(diversity(imgvr.rar,index="simpson"))
simpson$region <- imgvr.dat$region

p1 <- ggplot(simpson, aes(x=region,y=simpson$diversity,fill=region)) +
  geom_boxplot(show.legend=F,outlier.shape = NA) +
  geom_jitter(show.legend = F,shape=21) +
  ggtitle("Simpson diversity\nIMG/VR phage genomes\n2016") +
  theme(plot.title = element_text(lineheight=.8, face="bold", hjust = 0.5)) +
  xlab("") + 
  ylab("Index") +
  scale_fill_manual(values=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#FFFF33","#A65628") )
  # geom_text()))

# Richness: Chao1
# richness <- data.frame(estimateR(t(imgvr.rar))[2,])
richness <- data.frame(colSums(t(imgvr.rar > 0)))
richness$region <- imgvr.dat$region

p2 <- ggplot(richness, aes(x=region, y=richness$colSums,fill=region)) +
  geom_boxplot(show.legend=F, outlier.shape = NA) +
  geom_jitter(show.legend=F,shape=21) +
  ggtitle("Chao1 richness\nIMG/VR phage genomes\n2016") +
  theme(plot.title = element_text(lineheight=.8, face="bold", hjust = 0.5)) +
  xlab("") + 
  ylab("") +
  scale_fill_manual(values=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#FFFF33","#A65628"))

# Pileou evenness
S <- apply(imgvr.rar[,-1]>0,1,sum)
J <-  data.frame(diversity(imgvr.rar,index="simpson")/log(S))
max(J) # 0.1752689
min(J) # 0.1374599
# The lower the J (evenness) the more variation in abundance between taxa
J$region <- imgvr.dat$region

p3 <- ggplot(J, aes(x=region,y=J$diversity,fill=region)) +
  geom_boxplot(show.legend=F, outlier.shape = NA) +
  geom_jitter(show.legend=F,shape=21) +
  ggtitle("Pileou evenness\nIMG/VR phage genomes\n2016") +
  theme(plot.title = element_text(lineheight=.8, face="bold", hjust = 0.5)) +
  xlab("") + 
  ylab("") +
  scale_fill_manual(values=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#FFFF33","#A65628"))


## Plot all in one
pdf('plots/final/imgvr_boxplots_rarefied.pdf', onefile=TRUE ,width = 20)
vplayout <- function(x, y) viewport(layout.pos.row = x, layout.pos.col = y)
grid.newpage()
pushViewport(viewport(layout = grid.layout(1, 3))) # defines numbers of rows and columns
print(p1,vp = vplayout(1, 1))
print(p2,vp = vplayout(1, 2))
print(p3,vp = vplayout(1, 3))
dev.off()
```

Boxplots - Huge phages rarefied diversity, richness and evenness
```{r, eval=TRUE}
huge.dat <- read.delim("datasets/custom_id_fragmentCount.tsv", header=1,row.names=1)
levels(huge.dat$region) <- c("AF","AS","EU","ME","NA","OC","SA")

# Exclude non-numerical data
myvars <- names(huge.dat) %in% c("region")
huge.num <- huge.dat[!myvars]

# Convert to matrix 
huge.num <- as.matrix(huge.num)

# Rarefy data 
(raremax <- min(rowSums(huge.num))) #2302
huge.rar <- rrarefy(huge.num,sample=raremax)

### Boxplots ###
col <- c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#FFFF33","#A65628")

# Simpson
simpson <- data.frame(diversity(huge.rar,index="simpson"))
simpson$region <- huge.dat$region

p1 <- ggplot(simpson, aes(x=region,y=simpson$diversity,fill=region)) +
  geom_boxplot(show.legend=F,outlier.shape = NA) +
  geom_jitter(show.legend = F,shape=21) +
  ggtitle("Simpson diversity\nHuge phages\n2016") +
  theme(plot.title = element_text(lineheight=.8, face="bold", hjust = 0.5)) +
  xlab("") + 
  ylab("Index") +
  scale_fill_manual(values=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#FFFF33","#A65628") )
  # geom_text()))

# Richness 
richness <- data.frame(colSums(t(huge.rar > 0)))
richness$region <- huge.dat$region

# pdf("plots/diversity_richness/boxplot_custom_richness_virus.pdf",onefile=T)
p2 <- ggplot(richness, aes(x=region, y=richness$colSums,fill=region)) +
  geom_boxplot(show.legend=F, outlier.shape = NA) +
  geom_jitter(show.legend=F,shape=21) +
  ggtitle("Chao1 richness\nHuge phages\n2016") +
  theme(plot.title = element_text(lineheight=.8, face="bold", hjust = 0.5)) +
  xlab("") + 
  ylab("") +
  scale_fill_manual(values=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#FFFF33","#A65628"))

# Pileou evenness
S <- apply(huge.rar[,-1]>0,1,sum)
J <-  data.frame(diversity(huge.rar,index="simpson")/log(S))
max(J) # 0.6225347
min(J) # 0.2358951
J$region <- huge.dat$region

p3 <- ggplot(J, aes(x=region,y=J$diversity,fill=region)) +
  geom_boxplot(show.legend=F, outlier.shape = NA) +
  geom_jitter(show.legend=F,shape=21) +
  ggtitle("Pileou evenness\nHuge phages\n2016") +
  theme(plot.title = element_text(lineheight=.8, face="bold", hjust = 0.5)) +
  xlab("") + 
  ylab("") +
  scale_fill_manual(values=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#FFFF33","#A65628"))


## Plot all in one
pdf('plots/final/huge_boxplots_rarefied.pdf', onefile=TRUE ,width = 20)
vplayout <- function(x, y) viewport(layout.pos.row = x, layout.pos.col = y)
grid.newpage()
pushViewport(viewport(layout = grid.layout(1, 3)))
print(p1,vp = vplayout(1, 1))
print(p2,vp = vplayout(1, 2))
print(p3,vp = vplayout(1, 3))
dev.off()
```

